src/components/ProviderLogin.svelte

```diff
@@ -0,0 +1,139 @@
+
+  import DropDownMenu from '$components/DropDownMenu.svelte';
+  import BlockSection from '$components/BlockSection.svelte';
+  import Button from '$components/Button.svelte';
+  import { pageContent } from '$lib/stores/pageContentStore';
+  import { Network, allNetworks, networkToInfo, selectedNetwork } from '$lib/stores/networksStore';
+  import { getApi, updateConnectionStatus } from '$lib/polkadotApi';
+  import { fetchAccounts, storeProviderAccounts } from '$lib/stores/accountsStore';
+
+  let wsProvider: WsProvider;
+  let thisWeb3Enable: typeof web3Enable;
+  let thisWeb3Accounts: typeof web3Accounts;
+
+  let networks = allNetworks();
+
+  let selectedNetworkAsString: string = '';
+  let customNetwork: string = '';
+
+  let thisDotApi = defaultDotApi;
+  dotApi.subscribe((api) => (thisDotApi = api));
+
+  let selectedAccountAsString: string = '';
+  let providerAccounts: Record<string, string> = {};
+
+  let canConnect = false;
+  $: canConnect = selectedNetworkAsString !== '' && selectedAccountAsString !== '';
+
+  // We need to access the user's wallet to get the accounts
+  onMount(async () => {
+    // This must be in onMount because the extension requires that you have a window to attach to.
+    // Since this project is precompiled, there will be no window until onMount
+    const polkadotExt = await import('@polkadot/extension-dapp');
+    thisWeb3Enable = polkadotExt.web3Enable;
+    thisWeb3Accounts = polkadotExt.web3Accounts;
+  });
+
+  selectedNetwork.subscribe(async (network) => {
+    console.log('selectedNetwork changed to ' + network);
+    if (network != undefined && network != Network.NONE) {
+      let info = networkToInfo[network];
+      console.dir(info);
+      let endpoint = networkToInfo[network].endpoint;
+      console.log('endpoint: ' + endpoint);
+      try {
+        await getApi(endpoint?.toString() ?? '', thisDotApi, wsProvider);
+        await fetchAccounts(network, thisWeb3Enable, thisWeb3Accounts, thisDotApi.api as ApiPromise);
+        await updateConnectionStatus(thisDotApi.api as ApiPromise);
+      } catch (e) {
+        console.log(e);
+        alert(
+          `could not connect to ${
+            endpoint?.toString() || 'empty value'
+          }. Please enter a valid and reachable Websocket URL.`
+        );
+      }
+    }
+  });
+
+  storeProviderAccounts.subscribe((accounts) => {
+    console.log('storeProviderAccounts changed');
+    console.dir(accounts);
+    providerAccounts = Object.fromEntries(Object.entries(accounts).map(([key]) => [key, key]));
+  });
+
+  function connect() {
+    $isLoggedIn = true;
+    pageContent.dashboard();
+  }
+
+  function becomeProvider() {
+    pageContent.becomeProvider();
+  }
+
+  function networkChanged() {
+    console.log('networkChanged');
+    console.log('selectedNetworkAsString: ' + selectedNetworkAsString);
+    $selectedNetwork = selectedNetworkAsString as Network;
+  }
+
+  function accountChanged() {
+    console.log('accountChanged');
```

Comment:
lots of logs. if you like them, totally keep.

Reply:
This is just for development.  We can and should delete them later.